var/spack/repos/builtin/packages/vecmem/package.py

```diff
@@ -63,6 +63,11 @@ class Vecmem(CMakePackage, CudaPackage):
     # and we can choose between always depending on googletest, or using FetchContent
     # depends_on("googletest", type="test")
 
+    def setup_build_environment(self, env):
+        if self.spec.satisfies("+sycl"):
+            env.set("SYCLCXX", self.compiler.cxx)
+            env.set("SYCLFLAGS", "-fsycl")
```

Comment:
Is this flag specific to `oneapi`? Does `hipsycl` support it?

```suggestion
            if self.spec.satisfies("%oneapi"):
                env.set("SYCLFLAGS", "-fsycl")
```

Cf. `babelstream`:
https://github.com/spack/spack/blob/ba79542f3cdae25f7250f37139281bd84b662714/var/spack/repos/builtin/packages/babelstream/package.py#L268-L271

Reply:
Good point! Indeed hipsycl (now known as AdaptiveCpp) doesn't support this flag: https://github.com/AdaptiveCpp/AdaptiveCpp/blob/5cb8bf12a7c3375925055c3e1f32d7b57f7a3284/bin/acpp#L1977